tests/run-make/a-b-a-linker-guard/rmake.rs

```diff
@@ -1,44 +1,37 @@
 // ignore-tidy-linelength
 
+// Test that if we build `b` against a version of `a` that has one set of types, it will not run
+// with a dylib that has a different set of types.
+
 extern crate run_make_support;
 
-use run_make_support::{run, run_fail, rustc};
+use run_make_support::{run, run_fail, CodegenOpt, Rustc, SymbolManglingVersion};
+use CodegenOpt::*;
 
 fn main() {
-    rustc()
-        .arg("a.rs")
-        .arg("--cfg")
-        .arg("x")
-        .arg("-C")
-        .arg("prefer-dynamic")
-        .arg("-Z")
-        .arg("unstable-options")
-        .arg("-C")
-        .arg("symbol-mangling-version=legacy")
+    Rustc::new()
+        .input_file("a.rs")
+        .cfg("x")
+        .enable_unstable_options()
+        .codegen_opt(PreferDynamic)
+        .codegen_opt(SymbolManglingVersion(SymbolManglingVersion::Legacy))
         .run();
 
-    rustc()
-       .arg("b.rs")
-       .arg("-C")
-       .arg("prefer-dynamic")
-       .arg("-Z")
-       .arg("unstable-options")
-       .arg("-C")
-       .arg("symbol-mangling-version=legacy")
-       .run();
+    Rustc::new()
+        .input_file("b.rs")
+        .enable_unstable_options()
+        .codegen_opt(PreferDynamic)
+        .codegen_opt(SymbolManglingVersion(SymbolManglingVersion::Legacy))
```

Comment:
I'm not sure whether this really is better than what we had before (with merging the two like `-Cprefer-dynamic` into one line, which could be asserted by arg panicking when it sees -Z or -C).

Reply:
Don't really have a super strong opinion for the support library API atm, still experimenting *shrug*

It does hardcode the cli interface into the support library. But if the cli interface changes, then surely the tests affected will have to change with it anyway?

Reply:
Also I can omit the codegen_opt layer and just provide e.g. `prefer_dynamic()` / `lto(LtoKind::Fat)` directly.

Reply:
I think arg is still simpler, easier to maintain and more obvious to understand